### ISSUE TITLE
Enable Native Double Tap as Keybinding (keymap.cson)

### ISSUE THREAD

**Issue by rajmondx** [pulsar-edit/pulsar#301]
### Have you checked for existing feature requests?

- [X] Completed

### Summary

Allow users to double tap any button (eg. as IDEA allows it with their "double shift")
something like this

```cson
'.platform-win32':
  'shift-shift': 'fuzzy-finder:toggle-file-finder'

# or plus
'.platform-win32':
  'shift+shift': 'fuzzy-finder:toggle-file-finder'
```

### What benefits does this feature provide?

No big deal. I would just love to see this feature because of the muscle memory gained in IDEA, hard to not double press shift and somehow I believe some idea users might not like any ide if its not possible to configure double tap.

### Any alternatives?

https://gist.github.com/spaceribs/76b62d6cedb1d0892ec10a8b35d38683

```coffee
keyPressed = false;

atom.keymaps.addKeystrokeResolver ({event}) ->
  if event.key == "Shift" && event.type == "keydown"
    if keyPressed
      keyPressed = false;
      console.log('keyPressed');
      return 'shift-double';
    else
      keyPressed = true;
      setTimeout (-> keyPressed = false), 200;
      return null;
```

**Comment by confused-Techie** [comment]
Thanks for the contribution, this would be a pretty cool feature to see integrated

**Comment by rajmondx** [comment]
> Thanks for the contribution, this would be a pretty cool feature to see integrated

Check "Alternatives": I wrote a script which you could add to your own `init.js`, its a bit hacky and it could be done better (eg. if your `init.js` gets bigger such "little scripts" end up messy) but it does what its supposed to do.

**Comment by mauricioszabo** [comment]
I don't think this is supported by Electron or browsers in general (trigger an event when a modifier key is pressed). If they don't send us the keystroke, there's nothing we can actually do...

One thing they you CAN do already is double tap any key that's not a modifier. For example, the keystroke 'esc esc' works today (double tap escape key).

**Comment by savetheclocktower** [comment]
We do actually get told about all button presses, even those that don't produce characters. For instance, I have `pulsar-hover` configured to show hover information when I press `Opt+Ctrl`, and hide when I press it again.

Double-tapping the same modifier theoretically should work just like double-tapping another key (like `esc esc`). My guess is that modifiers take their own code path when interpreting keystrokes (and especially key sequences).

For someone that wants to add this feature properly: it'd probably have to go in the [`atom-keymap`](https://github.com/pulsar-edit/atom-keymap/) package. Add a new spec, then figure out how to get it to pass. As long as it doesn't introduce regressions, it'd probably get accepted.
